controllers/csm_controller.go

```diff
@@ -728,6 +730,29 @@ func (r *ContainerStorageModuleReconciler) PreChecks(ctx context.Context, cr *cs
 		return fmt.Errorf("unsupported driver type %s", cr.Spec.Driver.CSIDriverType)
 	}
 
+	// check for owner reference
+	deployments := r.K8sClient.AppsV1().Deployments(cr.Namespace)
+	driver, err := deployments.Get(ctx, cr.Name+"-controller", metav1.GetOptions{})
+	if err != nil {
+		log.Infow("Driver not installed yet")
+	} else {
+		if driver.GetOwnerReferences() != nil {
+			cred := driver.GetOwnerReferences()
+			for _, m := range cred {
+				if m.Name == cr.Name {
+					log.Infow("Owner reference is found and matches")
+					break
+				} else {
+					return fmt.Errorf("Required Owner reference not found. Please re-install driver ")
+				}
+			}
+
+		} else {
+			return fmt.Errorf("Owner reference not found. Please re-install driver")
```

Comment:
Please see my other comment. The customer will be stuck here. When they try to delete the CR and deploy a new driver, the operator will prevent the custom resource deletion as the finalizer will still be set.

Reply:
if we have a driver which doesn't have a owner reference, then we need to manually delete the driver by taking out finalizer and reinstalling the driver again. csm cannot help with deleting the driver if it doesn't recognize owner reference.